clairmeta/dcp_check_am.py

```diff
@@ -116,18 +117,55 @@ def check_assets_am_path(self, am, asset):
         """ AssetMap assets path validation.
 
             Reference :
-                SMPTE 429-9-2014 7.1
+                SMPTE 429-9-2014 7.1, A.1
         """
-        uuid, path, _ = asset
 
-        if path == '':
-            raise CheckException("Empty path for {}".format(uuid))
+        path_errors = []
+        _, path, _ = asset
+
+        path_segments = list(filter(None, path.split('/')))
+        path_segments_count = len(path_segments)
+        if path_segments_count > 10:
+            path_errors.append(">10 path segments: {}".format(path_segments_count))
 
-        if ' ' in path:
-            raise CheckException("Space in path")
+        max_path_seg = max(map(len, path_segments))
+        if max_path_seg > 100:
+            path_errors.append("Path segment >100 characters: {}".format(max_path_seg))
+
+        if len(path) > 100:
+            path_errors.append("Path >100 characters: {}".format(len(path)))
+
+        path_invalid_chars = re.findall(r'[^a-zA-Z0-9._/-]', path)
+        if path_invalid_chars:
+            path_errors.append("Invalid characters in path: {}".format(str(sorted(set(path_invalid_chars)))[1:-1]))
+
+        if path[0] == '/':
+            path_errors.append("Path is not relative")
+
+        if os.path.relpath(self.dcp.path + '/' + path, self.dcp.path).startswith("../"):
+            path_errors.append("Path points outside of DCP root")
```

Comment:
Although we don't support Windows explicitly (need to switch to Github Action for this), I wonder if that kind of explicit path manipulation would be safe (eg. using forward slash) ?

Reply:
I thinkt it is safe. This is because the XML type is xs:anyURI and also the documents mentions RFC 3986 (or UDF for Interop). I believe that it is only valid to have "UNIX-like" paths in the asset map.

An alternative would be to use the really nice Python [pathlib](https://docs.python.org/3/library/pathlib.html) but maybe it is overkill? It also requires Python 3.4.

Reply:
Thought about this some more. You are right. Will use os.path.join() instead.